src/diag.rs

```diff
@@ -95,6 +96,8 @@ pub enum Diagnostic {
     VariableRedeclaredAsConstant(TokenIndex, TokenAddress),
 }
 
+pub type Result<T> = result::Result<T, Diagnostic>;
```

Comment:
Not opposed, but unclear if this will see use outside verify.  Most places log an error out of band and then continue on.